mscore/plugin/api/excerpt.h

```diff
@@ -0,0 +1,96 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2019 Werner Schweer and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENCE.GPL
+//=============================================================================
+
+#ifndef __PLUGIN_API_EXCERPT_H__
+#define __PLUGIN_API_EXCERPT_H__
+
+#include "libmscore/excerpt.h"
+
+namespace Ms {
+
+namespace PluginAPI {
+
+class Score;
+
+//---------------------------------------------------------
+//   Excerpt
+//    Wrapper class for Excerpt
+//
+//   This is based on the wrapper in scoreelement.h, which
+//   we cannot use here, because Ms::Excerpt is not derived
+//   from Ms::ScoreElement.
+//   Since a plugin should never need to create an Excerpt
+//   instance by itself, we don't care for Ownership here.
+//---------------------------------------------------------
+
+class Excerpt : public QObject {
+    Q_OBJECT
+    Q_PROPERTY(Ms::PluginAPI::Score* partScore READ partScore)
+    Q_PROPERTY(QString               title     READ title)
+
+ protected:
+    Ms::Excerpt* const e;
+
+ public:
+    Excerpt(Ms::Excerpt* _e = nullptr)
+       : QObject(), e(_e) {}
+    Excerpt(const Excerpt&) = delete;
+    Excerpt& operator=(const Excerpt&) = delete;
+    virtual ~Excerpt() {};
+
+    Score* partScore();
+    QString title() { return e->title(); }
+};
+
+//---------------------------------------------------------
+//   wrap
+//---------------------------------------------------------
+template <class Wrapper, class T>
+Wrapper* excerptWrap(T* t)
+      {
+      Wrapper* w = t ? new Wrapper(t) : nullptr;
+      // All wrapper objects should belong to JavaScript code.
+      QQmlEngine::setObjectOwnership(w, QQmlEngine::JavaScriptOwnership);
+      return w;
+      }
+
+extern Excerpt* excerptWrap(Ms::Excerpt* e);
+
+//---------------------------------------------------------
+//   qml access to containers of Excerpt
+//
+//   QmlExcerptsListAccess provides a convenience interface
+//   for QQmlListProperty providing read-only access to
+//   plugins for Excerpts containers.
+//
+//   based on QmlListAccess in scoreelement.h
+//---------------------------------------------------------
+
+template <typename T, class Container>
+class QmlExcerptsListAccess : public QQmlListProperty<T> {
```

Comment:
Isn't the wrapper defined in scoreelement.h sufficient? It is not bound to ScoreElement in any way so you can just use it if needed. If you wish to avoid including `scoreelement.h` you can move that container wrapper to a separate header.

Reply:
Ah, I see, the ownership enum issue. Well, then a separate wrapper should probably work for now.

Reply:
I've created an alternative version of this in <a href="https://github.com/heuchi/MuseScore/tree/283179-plugin-api-noNewWrapper/">another branch</a>. Here direct links to <a href="https://github.com/heuchi/MuseScore/blob/283179-plugin-api-noNewWrapper/mscore/plugin/api/excerpt.h">excerpt.h</a> and <a href="https://github.com/heuchi/MuseScore/blob/283179-plugin-api-noNewWrapper/mscore/plugin/api/excerpt.cpp">excerpt.cpp</a> without a new wrapper. If you prefer that version, I can copy it here.

Reply:
No, I believe that is not necessary. Maybe the wrappers could be unified somehow but there is no need to add an unnecessary ownership variable just for that purpose.